Library/Homebrew/utils/fork.rb

```diff
@@ -33,10 +44,27 @@ def self.safe_fork(&_block)
             socket.close
           end
           write.close
-          data = read.read
+          # Each line on the error pipe contains a JSON-serialized exception.
+          # We read the first, since only one is necessary for a failure.
+          data = read.gets
           read.close
           Process.wait(pid) unless socket.nil?
-          raise Marshal.load(data) unless data.nil? || data.empty? # rubocop:disable Security/MarshalLoad
+
+          error_hash = JSON.parse(data) unless data.nil? || data.empty?
+
+          if error_hash
+            e = case error_hash["json_class"]
+            when "ErrorDuringExecution"
```

Comment:
Any way of turning this into an actual class comparison?

Reply:
Probably not at this point in the program -- we're comparing with strings because we're about to (re)create the classes that are coming from the error pipe in JSON form.